core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -52,12 +55,22 @@ class HadoopMapReduceCommitProtocol(jobId: String, path: String)
    */
   @transient private var addedAbsPathFiles: mutable.Map[String, String] = null
 
+  @transient private var partitionPaths: mutable.Set[String] = null
+
+  @transient private var stagingDir: Path = _
```

Comment:
Maybe faster?  We are not deleting the files one by one. We drop the whole staging directory.